pkg/adapter/v2/cloudevents.go

```diff
@@ -168,21 +173,28 @@ func (c *client) reportCount(ctx context.Context, event cloudevents.Event, resul
 
 		var res *http.Result
 		if !cloudevents.ResultAs(result, &res) {
-			return c.reportError(reportArgs, result)
+			c.reportError(reportArgs, result)
 		}
 
 		if rErr := c.reporter.ReportEventCount(reportArgs, res.StatusCode); rErr != nil {
-			// metrics is not important enough to return an error if it is setup wrong.
-			// So combine reporter error with ce error if not nil.
-			if result != nil {
-				result = fmt.Errorf("%w\nmetrics reporter error: %s", result, rErr)
+			fmt.Printf("metrics reporter error %s: %s \n", result, rErr)
+		}
+
+		if len(rres.Attempts) > 0 {
+			for _, retryResult := range rres.Attempts {
+				var res *http.Result
+				if !cloudevents.ResultAs(retryResult, &res) {
+					c.reportError(reportArgs, result)
+				}
+				if rErr := c.reporter.ReportRetryEventCount(reportArgs, res.StatusCode); rErr != nil {
+					fmt.Printf("metrics reporter error %s: %s\n", result, rErr)
```

Comment:
ditto.

Reply:
@vaikas ^ so you essentially want me to get rid of print statements?

Reply:
Yes, I'd prefer we use logging if you need to report this, so we actually get things like timestamps, etc. Printf just seems wonky here.